unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/FriendsController.cs

```diff
@@ -9,18 +10,24 @@ namespace DCL.Social.Friends
 {
     public class FriendsController : IFriendsController
     {
-        // TODO: remove singleton, rename to FriendsService and add to service locator
-        public static FriendsController i { get; private set; }
+        private const string USE_SOCIAL_CLIENT_FEATURE_FLAG = "use-social-client";
 
         private readonly IFriendsApiBridge apiBridge;
+        private readonly ISocialApiBridge socialApiBridge;
         private readonly Dictionary<string, FriendRequest> friendRequests = new ();
         private readonly Dictionary<string, UserStatus> friends = new ();
+        private readonly DataStore dataStore;
+
+        private CancellationTokenSource controllerCancellationTokenSource = new ();
+        private UniTaskCompletionSource featureFlagsInitializedTask;
+
+        private bool useSocialApiBridge => dataStore.featureFlags.flags.Get().IsFeatureEnabled(USE_SOCIAL_CLIENT_FEATURE_FLAG);
+        private FeatureFlag featureFlags => dataStore.featureFlags.flags.Get();
```

Comment:
This is redundant.

Reply:
`featureFlags` is being used in the class by already existinent code, modified `useSocialApiBridge` to depend on it